phoenix/app/cad_products_phoenix/lib/cad_products_phoenix_web/controllers/report_controller.ex

```diff
@@ -0,0 +1,14 @@
+defmodule CadProductsPhoenixWeb.ReportController do
+  use CadProductsPhoenixWeb, :controller
+
+  action_fallback CadProductsPhoenixWeb.FallbackController
+
+  @path "lib/cad_products_phoenix_web/reports/report_products.csv"
+
+  def create(conn, _) do
+    case Exq.enqueue(Exq, "report", CadProductsPhoenixWeb.Jobs.CreateReportJob, [@path]) do
+      {:ok, _} -> send_download(conn, {:file, @path})
```

Comment:
Se é assíncrono, como vai enviar o arquivo para download logo em seguida? Se o arquivo for grande, no momento desse envio o arquivo ainda estará sendo gerado